### DISCUSSION TITLE
Envoy health check actions

### DISCUSSION THREAD

**Discussion by lucasponce** [kiali/kiali#3816] [General]
In past Kiali Watercooler @oschaaf proposed that Kiali could enhance the troubleshooting capabilities of Envoy with some highlevel "health cheack action" that can be triggered/started from a Workload view.

This is an active area of development in Kiali as @xeviknal is working on features like
https://github.com/kiali/kiali/issues/3635

Where a user can show Envoy configuration in a friendly way and be linked with other Istio resources as VirtualServices and DestinationRules.

In parallel, @jshaughn is working on logs enhancements, in particular, to increase the coverage of Envoy logs in a way that can be correlated with other signals (telemetry/tracing) and be used to trigger specific actions.

I'm creating this discussion to unify all these discussions (watercooler, slack) into a single one where we can comment on the topic and give visibility.

Also, the kiali room on Istio has interesting discussions with users around same topic.

**Comment by jshaughn** [comment]
To summarize the log work, I guess there are two things going on with respect to the envoy access logs.  First, an integrated logs view that allows the proxy logs to be interspersed with app container logs.  And second, a parsed view of an individual access log (using engarde for parsing).  The user can see the parsed log in a tooltip, currently, but that is going to change soon to a modal, which will allow us a lot more flexibility to add related information, links, etc.
